src/braket/pennylane_plugin/translation.py

```diff
@@ -558,14 +560,21 @@ def translate_result_type(
             return DensityMatrix(targets)
         raise NotImplementedError(f"Unsupported return type: {return_type}")
 
-    if isinstance(measurement.obs, (Hamiltonian, qml.Hamiltonian)):
+    if isinstance(observable, (Hamiltonian, qml.Hamiltonian)):
         if return_type is ObservableReturnTypes.Expectation:
             return tuple(
-                Expectation(_translate_observable(term), term.wires) for term in measurement.obs.ops
+                Expectation(_translate_observable(term), term.wires) for term in observable.ops
             )
         raise NotImplementedError(f"Return type {return_type} unsupported for Hamiltonian")
 
-    braket_observable = _translate_observable(measurement.obs)
+    if return_type is ObservableReturnTypes.Sample and observable is None:
+        if isinstance(measurement, qml.measurements.SampleMeasurement):
+            return tuple(
+                Sample(BraketObservable.Z(), target) for target in targets or measurement.wires
+            )
+        raise NotImplementedError(f"Unsupported measurement type: {type(measurement)}")
```

Comment:
instead of embedding this logic here, would it be possible to have `_translate_observable` return both an observable and targets, and have it fall back to these defaults if `observable` is none and/or `targets` is none? Might be slightly cleaner (and then maybe wouldn't require the `C901` suppression above either, since the logic would be outside this function).

Reply:
I am not sure about that. The target cannot be specified if an observable is specified, so the branching will be based on whether the observable is None. Therefore, I believe that the overall logic will remain the same, and it will only be divided into more functions.